src/loader-xerces.cpp

```diff
@@ -208,6 +208,23 @@ void XercesSWIDTagIO::save(const string & filename, const SWIDStruct & what) {
 }
 
 
+validity XercesSWIDTagIO::is_xsd_valid(const std::string & filename) {
```

Comment:
this is not reentrant and I don't understand why. You always cerate a new parser so why not just create a parser for this method only?

Reply:
Xerces is very difficult to use properly due to very bad documentation. It is not obvious how to approach parsers, documents etc. I will look into that.